scripts/copy_ocp_aws_azure_data.py

```diff
@@ -0,0 +1,359 @@
+                drop_partitions(conn, schema, table_info["partable_name"])
+                get_or_create_partitions(conn, schema, table_info["partable_name"], partable_min_date)
+            except ProgrammingError as p:
+                LOG.warning(
+                    f"{p.__class__.__name__} :: {p}{os.linesep}Skip processing "
+                    + f"for {schema}.{table_info['basetable_name']}."
+                )
+                conn.rollback()
+                continue
+            except Exception as e:
+                conn.rollback()
+                LOG.warning(f"VERY WARNING :: {e.__class__.__name__} :: {e}")
+                continue
+            else:
+                try:
+                    conn.commit()
+                except Exception as x1:
+                    LOG.warning(
+                        f"{x1.__class__.__name__} :: {x1}{os.linesep}Skip processing "
+                        + f"for {schema}.{table_info['basetable_name']}."
+                    )
+                    conn.rollback()
+                    continue
+
+            try:
+                # copy data earlier than the threshold since it should be static.
+                copy_data(
+                    conn,
+                    schema,
+                    table_info["partable_name"],
+                    table_info["partable_cols"],
+                    table_info["basetable_name"],
+                )
+            except (ProgrammingError, ForeignKeyViolation) as p:
+                LOG.warning(
+                    f"{p.__class__.__name__} :: {p}{os.linesep}Rolling back copy transaction "
+                    + f"for {schema}.{table_info['basetable_name']}."
+                )
+                conn.rollback()
+            except Exception as e:
+                conn.rollback()
+                LOG.warning(f"VERY WARNING :: {e.__class__.__name__} :: {e}")
+            else:
+                try:
+                    conn.commit()
+                except Exception as x1:
+                    LOG.warning(
+                        f"{x1.__class__.__name__} :: {x1}{os.linesep}Rollback copy transaction on COMMIT exception "
+                        + f"for {schema}.{table_info['basetable_name']}."
+                    )
+                    conn.rollback()
+
+
+def main():
+    # get customer schemata
+    schemata = tables = None
+    with connect() as conn:
+        schemata = get_customer_schemata(conn)
+        tables = get_ocpawsazure_tables(conn)
+
+    if schemata:
+        # Combine for the individual job work
+        target_tables = [{"schema": schema, "table_info": table_info} for schema in schemata for table_info in tables]
+        t_tot = len(target_tables)
+        del schemata
+        del tables
+
+        # start worker processes
+        max_workers = int(sys.argv[1]) if len(sys.argv) > 1 else int(os.environ.get("NUM_JOB_WORKERS", 1))
+        if max_workers > t_tot:
+            max_workers = t_tot
```

Comment:
Question: Do we _need_ to run this script with multiple workers?

Reply:
Why not? It's an attempt to make such a potentially long job faster. Each table copy is discrete. Everything is logged by the process.